### ISSUE TITLE
Allow logging to be configured in lab chart (debugging only)

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#2722]
When debugging a kafka issue it was necessary to add logging to the lab helm chart

This issue allows for this configuration, though the default remains off

Also uplifts kafka from 2.3 to 2.4

**Comment by mandy-chessell** [comment]
This suggests actually that we need to improve the audit logging of the kafka connector - we seem to have lost the ability to detect problems with the kafka set up - such as
* Kakfa is down
* The topics are not defined (and auto-define is not allowed)
* The server does not have access to the topic.
Turning on debug is not the right answer - unless as an interim step to identify the places where audit logging is needed.

Bottom line is that it should be possible to resolve set up problems with kafka from the audit log.  Debug logging should only be turned on to debug bugs in kafka.

**Comment by planetf1** [comment]
We're learning more about kafka behaviour in https://github.com/odpi/egeria/issues/2717 & as you say need to incrementally improve the audit logging to support ffdc in the field.

I've also requested further audit logging in a related area - of cohort initialization (not kafka)in #2596

The proposal here is two fold
 a) To bring kafka server (the dependent chart we use) to 2.4 -- I observed some client/server mismatched (handled fine, as kafka is meant to do, but acted as a reminder since dependabot doesn't catch these)

b) To /enable/ the ability to debug in the k8s charts. this was in fact the only place we could adequately debug as logging impacts performance (which is exactly why it's really not a good thing to have to do).  But it was needed to debug, by a developer, the problem. The default is unchanged, indeed unspecified, it will be audit logs only, but it means a configuration override can modify that behaviour. The developer can do this easily when running standalone in a local environment since spring will pass environment variables or allow properties to be overriden. This is not the case here, which is why the environment entry was made explicit. It is needed for all components, since we don't know which ones in advance, and audit log entries are not suitable for this *debug* purpose - as one can see from the referenced defect.

An alternative proposal for b) would be to allow an environment map to be passed (i think this is possible) which would allow any variables to be passed through though this is additional complexity. This is better as it is more general purpose of course so wouldn't be restricted

I have no doubt though that in some circumstances this is needed, and it's better to have in place than try and modify the helm templates or k8 resources at the time. It would comprimise the ability to debug (by developers) too much to omit [but see alternative... ]

**Comment by planetf1** [comment]
There's some more specific suggestions of what needs addressing in our kafka tests -- and audit log reporting at https://github.com/odpi/egeria/issues/1876 to ensure that all the scenarios we both mentioned are clear to the administrator via the audit log.

(This goes beyond kafka too)

That was closed but #2709 was opened for a re-implemented connector - which does need the comments from 2709 moved in.

I see those (and other raised kafka issues) as pushing forward with the end user serviceability of our communication stack.

**Comment by planetf1** [comment]
No action needed at this point. Will revisit when further debugging needed and current chart facilities are insufficient
